### PR TITLE
badwolf: 1.2.2 -> 1.3.0

### PR DESCRIPTION
**Pull request by cu1ch3n**
## Description of changes

Changelog: https://hacktivis.me/releases/badwolf-1.3.0.txt

Related issue: https://github.com/NixOS/nixpkgs/issues/269135

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/browsers/badwolf/default.nix [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
   pname = "badwolf";
-  version = "1.2.2";
+  version = "1.3.0";
 
-  src = fetchgit {
-    url = "git://hacktivis.me/git/badwolf.git";
-    rev = "v${version}";
-    hash = "sha256-HfAsq6z+1kqMAsNxJjWJx9nd2cbv0XN4KRS8cYuhOsQ=";
+  src = fetchurl {
+    url = "https://hacktivis.me/releases/${pname}-${version}.tar.gz";
```
Comment (Aleksanaa) [inline pkgs/applications/networking/browsers/badwolf/default.nix:15]:
```suggestion
    url = "https://hacktivis.me/releases/badwolf-${version}.tar.gz";
```

File: pkgs/applications/networking/browsers/badwolf/default.nix [hunk 2]
```diff
@@ -1,38 +1,37 @@
...
+    hash = "sha256-J238y6it38IFzrEEd2aOSytqSFPzRMhtXB41sccDRZ8=";
   };
 
   preConfigure = ''
```
Comment (Aleksanaa) [inline pkgs/applications/networking/browsers/badwolf/default.nix:19]: Just use `prefixKey = "PREFIX=";`

File: pkgs/applications/networking/browsers/badwolf/default.nix [hunk 3]
```diff
@@ -1,38 +1,37 @@
...
     export PREFIX=$out
   '';
 
-  nativeBuildInputs = [ pkg-config wrapGAppsHook ];
+  dontAddPrefix = true;
 
-  buildInputs = [ webkitgtk libxml2 gettext glib glib-networking ];
+  nativeBuildInputs = [ pkg-config ninja ed ];
```
Comment (Aleksanaa) [inline pkgs/applications/networking/browsers/badwolf/default.nix:25]: `wrapGAppsHook` should still be needed.

File: pkgs/applications/networking/browsers/badwolf/default.nix [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
+
+  buildInputs = [ webkitgtk libxml2 ];
 
   meta = with lib; {
     description = "Minimalist and privacy-oriented WebKitGTK+ browser";
     homepage = "https://hacktivis.me/projects/badwolf";
     license = licenses.bsd3;
     platforms = platforms.linux;
```
Comment (Aleksanaa) [inline pkgs/applications/networking/browsers/badwolf/default.nix:33]: Please also add `mainProgram`.

File: pkgs/applications/networking/browsers/badwolf/default.nix [hunk 5]
```diff
@@ -1,38 +1,37 @@
...
 
-  buildInputs = [ webkitgtk libxml2 gettext glib glib-networking ];
+  nativeBuildInputs = [ pkg-config ninja ed ];
+
+  buildInputs = [ webkitgtk libxml2 ];
 
   meta = with lib; {
     description = "Minimalist and privacy-oriented WebKitGTK+ browser";
     homepage = "https://hacktivis.me/projects/badwolf";
     license = licenses.bsd3;
```
Comment (Aleksanaa) [inline pkgs/applications/networking/browsers/badwolf/default.nix:32]: Upstream also ships `cc-by-sa-40` in their repository. See https://hacktivis.me/git/badwolf/file/LICENSES/CC-BY-SA-4.0.txt.html

**Comment by Aleksanaa** [comment]
Please use `Closes https://github.com/NixOS/nixpkgs/issues/269135` to avoid it being ignored.

**Comment by Aleksanaa** [comment]
Hi, do you still want to work on this?

**Comment by cu1ch3n** [comment]
@Aleksanaa Hi! Sorry for the late response. I have been busy with other projects. And I don't have the local repo right now. Would you like to help me modify it according to your changes and remove me from the maintainers of this package? Thanks!

**Comment by Aleksanaa** [comment]
Opened new PR here: https://github.com/NixOS/nixpkgs/pull/300739

**Pull request closed without merging**